src/realm/sync/protocol.hpp

```diff
@@ -206,13 +206,26 @@ struct ResumptionDelayInfo {
 };
 
 struct ProtocolErrorInfo {
+    enum class Action {
+        NoAction,
```

Comment:
I had an `Optional` initially, but since we expose it to the user and we cannot use `Optional`s in the c api (we could use a pointer, but it would be wrong IMO), I think this is a good compromise.